projects/swimlane/ngx-charts/src/lib/line-chart/line-series.component.ts

```diff
@@ -120,7 +120,40 @@ export class LineSeriesComponent implements OnChanges {
   }
 
   getLineGenerator(): any {
-    return line<any>()
+    if (this.curve == curveCatmullRom) {
+      return line<any>()
+      .x(d => {
+        const label = d.name;
+        let value;
+        if (this.scaleType === ScaleType.Time) {
+          value = this.xScale(label);
+        } else if (this.scaleType === ScaleType.Linear) {
+          value = this.xScale(Number(label));
+        } else {
+          value = this.xScale(label);
+        }
+        return value;
+      })
+      .y(d => this.yScale(d.value))
+      .curve(this.curve.alpha(this.tension));
```

Comment:
Indent to match coding style
```suggestion
        .x(d => {
          const label = d.name;
          let value;
          if (this.scaleType === ScaleType.Time) {
            value = this.xScale(label);
          } else if (this.scaleType === ScaleType.Linear) {
            value = this.xScale(Number(label));
          } else {
            value = this.xScale(label);
          }
          return value;
        })
        .y(d => this.yScale(d.value))
        .curve(this.curve.alpha(this.tension));
```